File: src/main/java/com/oriaxx77/oaths/rest/OathResource.scala
package com.oriaxx77.oaths.rest

import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

import com.oriaxx77.oaths.domain.enity.Person
import org.springframework.web.bind.annotation.RequestMethod
import play.api.libs.json._
import collection.JavaConversions._
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.beans.factory.annotation.Autowire
import org.springframework.beans.factory.annotation.Autowired
import com.oriaxx77.oaths.domain.repository.OathRepository
import play.api.libs.json.Writes
import play.api.libs.json._
import com.oriaxx77.oaths.domain.enity.Oath
import com.oriaxx77.oaths.domain.enity.User
import Json.toJson
import scala.collection.JavaConverters._
import com.oriaxx77.oaths.rest.conversion.Jsonize
import org.springframework.web.bind.annotation.RequestHeader
import com.oriaxx77.oaths.domain.repository.UserRepository

@RestController
@RequestMapping(Array("/oaths"))
class OathResource @Autowired() (oathRepository: OathRepository,
    userRepository: UserRepository) {


  @RequestMapping(method = Array(RequestMethod.GET))
  def getOaths(): String = {
    import Jsonize._
    return Json.obj( "oaths" -> oathRepository.findAll().asScala).toString()
  }
  
  // TODO Integrate spring security
  @RequestMapping( value=Array("/mine"), method=Array(RequestMethod.GET) )
  def getMyOaths( /*@RequestHeader authToken: String*/ ): String = {
    import Jsonize._
    val user = userRepository.findAll().headOption.get // TODO Spring Security
    return Json.obj( "oaths" -> oathRepository.findByOathTaker( user ).asScala).toString() 
  }
  
  // TODO Integrate spring security
  @RequestMapping( value=Array("/allbutmine" ), method=Array(RequestMethod.GET))
  def getAllOathsButMine( /*@RequestHeader authToken: String*/ ): String = {
    import Jsonize._
    val user = userRepository.findAll().headOption.get // TODO Spring Security
    return Json.obj( "oaths" -> oathRepository.findByOathTakerNot( user ).asScala).toString()
  }

}

File: src/main/java/com/oriaxx77/oaths/rest/OathResource.java
package com.oriaxx77.oaths.rest;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import com.oriaxx77.oaths.domain.enity.Oath;
import com.oriaxx77.oaths.domain.enity.User;
import com.oriaxx77.oaths.domain.repository.OathRepository;
import com.oriaxx77.oaths.domain.repository.UserRepository;

@RestController
@RequestMapping("/oaths")
public class OathResource 
{

	private OathRepository oathRepository;
	private UserRepository userRepository;

	@Autowired
	public OathResource(OathRepository oathRepository, UserRepository userRepository) 
	{
		this.oathRepository = oathRepository;
		this.userRepository = userRepository;
	}

	@RequestMapping( method={RequestMethod.GET}, produces={MediaType.APPLICATION_JSON_VALUE})
	public Iterable<Oath> getOaths() 
	{
		return oathRepository.findAll();
	}

	// TODO Integrate spring security
	@RequestMapping(value = "/mine", method = { RequestMethod.GET }, produces = { MediaType.APPLICATION_JSON_VALUE })
	public Iterable<Oath> getMyOaths( /* @RequestHeader authToken: String */ ) 
	{
		User user = userRepository.findAll().iterator().next(); // TODO Spring// Security
		return oathRepository.findByOathTaker(user);
	}

	// TODO Integrate spring security
	@RequestMapping(value = "/allbutmine", method = { RequestMethod.GET }, produces = {MediaType.APPLICATION_JSON_VALUE })
	public Iterable<Oath> getAllOathsButMine( /*
												 * @RequestHeader authToken:
												 * String
												 */ ) 
	{
		User user = userRepository.findAll().iterator().next(); // TODO Spring// Security
		return oathRepository.findByOathTakerNot(user);
	}

}